### ISSUE TITLE
Block action when a yanked runtime version is used

### ISSUE THREAD

**Issue by jedcunningham** [astronomer/astro-cli#1642]
- [x] I have checked that a similar [feature request](https://github.com/astronomer/astro-cli/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.

**✍️ Is your feature request related to a problem? Please describe.**

Today, you can use any version of runtime ever released. However, some are known to be bad and broken.

**🧩 Describe the solution you'd like**

I'd love to see us block (with a flag to force past), or at least very noisily warn, when a yanked runtime image is used.

"Yanked" is a new concept for runtime, but there are new `yanked` and `yankedReason` flags that you can use to determine if a particular version is yanked and why.

**Is your feature request specific to a particular Astronomer Platform?**

No, this should apply everywhere I believe.

**Comment by sunkickr** [comment]
@paolaperaza should this only be blocked if an Org is not already using the version?

**Comment by cmarteepants** [comment]
> @paolaperaza should this only be blocked if an Org is not already using the version?

No, it should be generally blocked, similar to creating deployments with yanked versions. Whenever we yank an image, we will be notifying people that are currently using the yanked version and strongly encouraging them to upgrade. We are only yanking versions in exceptional circumstances where people should really not be running that version.
